### PR TITLE
ToDoLists

### PR DESCRIPTION
**Pull request by kehtolaulu**

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
-const { Schema, model, Types } = require('mongoose');
+const { Schema, model, Types } = require("mongoose");
+const ToDoList = require("./ToDoList").schema;
 
 const schema = new Schema({
     username: { type: String, required: true, unique: true },
     password: { type: String, required: true },
-    todos: [{ type: Types.ObjectId, ref: 'ToDo' }]
-})
+    toDos: [{ type: Types.ObjectId, ref: "ToDo" }],
```
Comment (Dimanaux) [inline models/User.js:7]: ne ne ne

File: routes/toDoLists.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    } catch (e) {
+        res.status(500);
+    }
+});
+
+router.delete("/:id", auth, async (req, res) => {
+    try {
+        let user = await User.findOne({ username: req.user.username });
+        let toDoLists = user.toDoLists;
+        user.toDoLinsts = toDoLists.filter(list => list.id !== req.params.id)
```
Comment (Dimanaux) [inline routes/toDoLists.js:48]: ;

File: routes/toDoLists.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        res.status(200);
+    } catch (e) {
+        res.status(500);
+    }
+});
+
+router.delete("/:id", auth, async (req, res) => {
+    try {
+        let user = await User.findOne({ username: req.user.username });
+        let toDoLists = user.toDoLists;
```
Comment (Dimanaux) [inline routes/toDoLists.js:47]: Do you really need this variable?

File: routes/toDoLists.js (hunk 2 above)
Comment (Dimanaux) [reply routes/toDoLists.js:48]: Add rule to Eslint

File: routes/toDoLists.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        user.toDoLinsts = toDoLists.filter(list => list.id !== req.params.id)
+        user.save();
+        res.status(200);
+    } catch (e) {
+        res.status(500);
```
Comment (Dimanaux) [inline routes/toDoLists.js:52]: ne ne ne

File: routes/toDoLists.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        res.json(toDoList.toDos);
+    } catch (e) {
+        res.status(500).json({ message: "Something went wrong" });
+    }
+});
+
+router.post("/:id/todos", auth, async (req, res) => {
+    try {
+        let toDoList = await ToDoList.findOne({ _id: req.params.id });
+        let date = Date.now();
```
Comment (Dimanaux) [inline routes/toDoLists.js:68]: Can it be default value declared in schema?

File: routes/toDoLists.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        let status = "new";
```
Comment (Dimanaux) [inline routes/toDoLists.js:69]: The same

File: routes/toDoLists.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        let text = req.body.text;
+        let todo = new ToDo({
+            text, status, date, owner: req.user.userId
+        });
+        await todo.save();
+        toDoList.toDos.push(todo);
+        toDoList.save();
```
Comment (Dimanaux) [inline routes/toDoLists.js:76]: Why await two lines above and no await here?

File: routes/toDoLists.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+});
+
+router.delete("/:id/todos/:todoId", auth, async (req, res) => {
+    try {
+        let toDoList = await ToDoList.findOne({ _id: req.params.id });
+        toDoList.toDos = toDoList.toDos.filter(todo => todo.id !== req.params.todoId);
+        toDoList.save();
+        res.status(200);
+    } catch (e) {
+        res.status(500);
```
Comment (Dimanaux) [inline routes/toDoLists.js:102]: I think this `try-catch` is useless. I think if exception happens then 500 will be returned automatically.

File: routes/toDoLists.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+router.put("/:id/todos/:todoId", auth, async (req, res) => {
+    try {
+        let toDoList = await ToDoList.findOne({ _id: req.params.id });
+        let toDo = toDoList.toDos.filter(todo => todo.id === req.params.todoId)[0];
+        toDo.status = req.body.status;
+        toDoList.save();
+        res.status(200);
+    } catch (e) {
+        res.status(500);
```
Comment (Dimanaux) [inline routes/toDoLists.js:91]: I think this try-catch is useless. I think if exception happens then 500 will be returned automatically.

File: models/User.js [hunk 10]
```diff
@@ -1,9 +1,10 @@
-const { Schema, model, Types } = require('mongoose');
+const { Schema, model, Types } = require("mongoose");
```
Comment (kehtolaulu) [inline models/User.js:1]: **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
'Types' is assigned a value but never used. [no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)

**Pull request merged**